src/Mollie/WC/Plugin.php

```diff
@@ -440,43 +440,69 @@ public static function enqueueComponentsAssets()
     }
 
     /**
-     * Payment return url callback
+     * Returns the order from the Request first by Id, if not by Key
+     *
+     * @return bool|WC_Order
      */
-    public static function onMollieReturn ()
+    public static function orderByRequest()
     {
-        $dataHelper = self::getDataHelper();
+        $dataHelper = getDataHelper();
 
         $orderId = filter_input(INPUT_GET, 'order_id', FILTER_SANITIZE_NUMBER_INT) ?: null;
         $key = filter_input(INPUT_GET, 'key', FILTER_SANITIZE_STRING) ?: null;
         $order = $dataHelper->getWcOrder($orderId);
 
         if (!$order) {
-            self::setHttpResponseCode(404);
-            self::debug(__METHOD__ . ":  Could not find order $orderId.");
-            return;
+            $order = $dataHelper->getWcOrder(wc_get_order_id_by_order_key($key));
+        }
+
+        if (!$order) {
+            throw new \RuntimeException(
```

Comment:
Sorry I missed this, better to import the class instead of define the global namespace